src/resolve-dependency.ts

```diff
@@ -47,35 +54,40 @@ function resolveDir (path, parent, job) {
   return resolveFile(resolve(path, 'index'), parent, job);
 }
 
-function notFound (specifier, parent) {
-  const e = new Error("Cannot find module '" + specifier + "' loaded from " + parent);
-  e.code = 'MODULE_NOT_FOUND';
-  throw e;
+class NotFoundError extends Error {
+  public code: string;
+  constructor(specifier: string, parent: string) {
+    super("Cannot find module '" + specifier + "' loaded from " + parent);
+    this.code = 'MODULE_NOT_FOUND';
+  }
 }
```

Comment:
I changed the `notFound()` function to a class because typescript doesn't keep track if a function throws or not so we need to throw this custom error where appropriate.

Reply:
An alternative might be make the return value `never` I believe.